python/cudf/cudf/core/buffer/utils.py

```diff
@@ -71,6 +73,14 @@ def as_buffer(
             "`data` is a buffer-like or array-like object"
         )
 
+    if get_option("copy_on_write"):
+        if isinstance(data, Buffer) or hasattr(
+            data, "__cuda_array_interface__"
+        ):
+            return CopyOnWriteBuffer._from_device_memory(data, exposed=exposed)
+        if exposed:
+            raise ValueError("cannot created exposed host memory")
+        return CopyOnWriteBuffer._from_host_memory(data)
     if get_global_manager() is not None:
```

Comment:
Was unable to make this change, since there were a bunch of pytests failing, probably because spilling could be enabled by setting a spilling manager too. 

cc: @madsbk 

Should we tackle this in a separate PR?